##### $N$ = number of elements
##### $\overline{X}$ = Mean of X
##### $\overline{Y}$ = Mean of Y

##### Sum of squares Definition formula (ss)
$SS = \sum(X - \overline{X})^2$

##### Sum of squares Computation formula (ss)
${SP}_{xy} = \Sigma{XY} - \frac{(\Sigma{X})(\Sigma{Y})}{n}$

$SS_{x} = \sum{X^2} - {\dfrac{(\sum{X})^2}{n}}$

$SS_{y} = \sum{Y^2} - {\dfrac{(\sum{Y})^2}{n}}$

$ r = \frac{SP_{xy}}{\sqrt{SS_xSS_y}}$

$b = r\sqrt{\dfrac{SS_y}{SS_x}}$

$a = \overline{Y} - b\overline{X}$

$Y' = b\overline{X} + a$

##### Variance
$\sigma^2 = \frac{SS}{N}$

##### Standarad Deviation
$\sigma = \sqrt{\frac{SS}{N}}$

##### Variance for Sample
$\sigma^2 = \frac{SS}{N - 1}$

##### Standarad Deviation for Sample
$\sigma = \sqrt{\frac{SS}{N - 1}}$

In [1]:
import math
import statistics
X = [13,9,7,5,1]
Y = [14,18,12,10,6]
friend = ['Doris','Steve','Mike','Andrea','John']

n = len(X)
sqx = [x**2 for x in X]
sqy = [y**2 for y in Y]
xy = [X[n]*Y[n] for n in range(0, len(X))]
sumx = sum(X)
sumy = sum(Y)
sumxy = sum(xy)
sum_sqx = sum(sqx)
sum_sqy = sum(sqy)
spxy = sumxy - (sumx*sumy)/n
ssx = sum_sqx - sumx**2/n
ssy = sum_sqy - sumy**2/n
r = spxy/math.sqrt(ssx*ssy)
mean_x = statistics.mean(X)
mean_y = statistics.mean(Y)
var_x = [(x-mean_x) for x in X]
var_y = [(y-mean_y) for y in Y]

print(var_x)
print(var_y)
print(f'mean_x = {mean_x}\nmean_y = {mean_y}')
print(f'spxy = {spxy}\nssx = {ssx}\nssy = {ssy}\nr = {r}')
print(f'sqx = {sqx}\nsqy = {sqy}')

[6, 2, 0, -2, -6]
[2, 6, 0, -2, -6]
mean_x = 7
mean_y = 12
spxy = 64.0
ssx = 80.0
ssy = 80.0
r = 0.8
sqx = [169, 81, 49, 25, 1]
sqy = [196, 324, 144, 100, 36]


In [2]:
b = r*(math.sqrt(ssx/ssy))
# Taking [7,12] from the list, as Xbar and Ybar
a = 12 - (b*7)
# a = 6 - (b*1)
print(a,b)
# Y_prime = b*X + a
# Y_prime = (0.8)X + 6.3999999999999995

6.3999999999999995 0.8


### Progress Check

In [3]:
# EDUCATIONAL LEVEL (X)
# X = 13
# SSx = 25
# WEEKLY READING TIME (Y)
# Y = 8
# SSy = 50
# r = .30

In [4]:
X = 13
Y = 8
b = 0.3*(math.sqrt(50/25))
a = Y - (b*X)
print(a,b)

2.4845671067449295 0.4242640687119285


In [5]:
# (a) Determine the least squares equation for predicting weekly reading time from educational level.
# Answer:
# Y' = 0.4242640687119285(X) + 2.4845671067449295

# (b) Faith’s education level is 15. What is her predicted reading time?
# Answer:
Y_prime1 = 0.4242640687119285*15 + 2.4845671067449295
print(Y_prime1)

# (c) Keegan’s educational level is 11. What is his predicted reading time?
# Answer:
Y_prime2 = 0.4242640687119285*11 + 2.4845671067449295
print(Y_prime2)

8.848528137423857
7.151471862576143


##### Table 7.2
PREDICTED CARD
RETURNS (Y ′) FOR
DIFFERENT CARD
INVESTMENTS (X )

##### Sum of squares Definition formula (ss)
$SS = \sum(X - \overline{X})^2$

##### Sum of squares Computation formula (ss)
$SS = \sum{X^2} - {\dfrac{(\sum{X})^2}{N}}$

$ r = \frac{SP_{xy}}{\sqrt{SS_xSS_y}}$

$b = r\sqrt{\dfrac{SS_y}{SS_x}}$

$a = \overline{Y} - b\overline{X}$

$Y' = b\overline{X} + a$

$Y' = .80(X) + 6.40$
  
$  = .80(1) + 6.40$
  
$  =  7.20 $

X = [13,9,7,5,1]

Y = [14,18,12,10,6]
##### Based on the mean of Y = 12; (x - 12)
$SS_y = (-6)^2 + (-2)^2 + 0^2 + 6^2 + 2^2 = 80$
##### Based on the regression line or the slope
$SS_{y|x} = (-1.2)^2 + (-0.4)^2 + (0)^2 +(4.4)^2 + (-2.8)^2 = 28.8$

##### Computation Breakdown

r = 64/(sqrt(80*80))

r = 0.8

mean_x = 7

mean_y = 12

a = Ybar - bXbar

a = 12 - (0.8*7) = 6.4

b = 0.8*(sqrt(80/80))

b = 0.8

Y' = bXbar + a

(1,6)
Y' = (0.8*1) + 6.4 = 7.2

6 - 7.2 = −1.2

(5,10)
Y' = (0.8*5) + 6.4 = 10.4

10 - 10.4 = −0.4

(7,12)
Y' = (0.8*7) + 6.4 = 12

12 - 12 = 0

(9,18)
Y' = (0.8*9) + 6.4 = 13.6

18 - 13.6 = 4.4

(13,14)
Y' = (0.8*13) + 6.4 = 16.8

14 - 16.8 = −2.8

##### percent accuracy $= {r}^2$

### Sum of Squares

In [6]:
import statistics
import math

friend = ['Doris','Steve','Mike','Andrea','John']
X = [13,9,7,5,1]
Y = [14,18,12,10,6]

n = len(X)
sqx = [x**2 for x in X]
sqy = [y**2 for y in Y]
xy = [X[n]*Y[n] for n in range(0, len(X))]
sumx = sum(X)
sumy = sum(Y)
sumxy = sum(xy)
sum_sqx = sum(sqx)
sum_sqy = sum(sqy)
spxy = sumxy - (sumx*sumy)/n
ssx = sum_sqx - sumx**2/n
ssy = sum_sqy - sumy**2/n
r = spxy/math.sqrt(ssx*ssy)
b = r*math.sqrt(ssy/ssx)
mean_x = statistics.mean(X)
mean_y = statistics.mean(Y)
a = mean_y - (b*mean_x)

print(f'spxy = {spxy}\nsqx = {sqx}\nsqy = {sqy}\nxy = {xy}')
print(f'sum_sqx = {sum_sqx}\nsum_sqy = {sum_sqy}')
print(f'sumx = {sumx}\nsumy = {sumy}\nsumxy = {sumxy}')
print(f'mean_x = {mean_x}\nmean_y = {mean_y}\nssx = {ssx}\nssy = {ssy}\nr = {r}\nb = {b}\na = {a}')

Y_prime = [Y[n]-((b*X[n])+a) for n in range(0, len(X))]
Y_prime_squared = [y**2 for y in Y_prime]
sum_of_squares = sum(Y_prime_squared)
print(f'Y_prime_squared = {Y_prime_squared}')
print(sum_of_squares)

sum_of_squares_based_on_mean = sum([(y - mean_y)**2 for y in Y])
sum_of_squares_based_on_mean

percent_accuracy = (sum_of_squares_based_on_mean - sum_of_squares)/sum_of_squares_based_on_mean
percent_accuracy

spxy = 64.0
sqx = [169, 81, 49, 25, 1]
sqy = [196, 324, 144, 100, 36]
xy = [182, 162, 84, 50, 6]
sum_sqx = 325
sum_sqy = 800
sumx = 35
sumy = 60
sumxy = 484
mean_x = 7
mean_y = 12
ssx = 80.0
ssy = 80.0
r = 0.8
b = 0.8
a = 6.3999999999999995
Y_prime_squared = [7.840000000000004, 19.360000000000003, 0.0, 0.15999999999999887, 1.4399999999999984]
28.800000000000004


0.6399999999999999

### Progress Check *7.4 

In [7]:
# Assume that an r of .30 describes the relationship between educational level and estimated hours spent reading each week.
r = 0.30

# (a) According to r squared, what percent of the variability in weekly reading time is predictable from its relationship with educational 
# level?
# Answer:
r_squared = r**2
print(r_squared*100)

# (b) What percent of variability in weekly reading time is not predictable from this relationship?
# Answer:
print(100 - (r_squared*100))

# (c) Someone claims that 9 percent of each person’s estimated reading time is predictable from the relationship. 
# What is wrong with this claim?

9.0
91.0
